extensions/mapstore/src/test/java/com/hazelcast/mapstore/MappingHelperTest.java

```diff
@@ -149,4 +150,87 @@ public void when_loadColumnMetadataFromMapping_then_escapeMappingName() {
         );
     }
 
+    @Test
+    public void tableName_to_externalName_simple() {
+        String tableName = "my_table";
+        String actual = MappingHelper.externalName(tableName);
+        Assertions.assertThat(actual).isEqualTo("\"my_table\"");
+    }
+
+    @Test
+    public void tableName_to_externalName_with_dot() {
+        String tableName = "custom_schema.my_table";
+        String actual = MappingHelper.externalName(tableName);
+        Assertions.assertThat(actual).isEqualTo("\"custom_schema\".\"my_table\"");
+    }
+
+    @Test
+    public void tableName_to_externalName_with_hyphen() {
+        String tableName = "schema-with-hyphen.table-with-hyphen";
+        String actual = MappingHelper.externalName(tableName);
+        Assertions.assertThat(actual).isEqualTo("\"schema-with-hyphen\".\"table-with-hyphen\"");
+    }
+
+    @Test
+    public void tableName_to_externalName_with_space() {
+        String tableName = "schema with space.table with space";
+        String actual = MappingHelper.externalName(tableName);
+        Assertions.assertThat(actual).isEqualTo("\"schema with space\".\"table with space\"");
+    }
+
+    @Test
+    public void tableName_to_externalName_with_2dots() {
+        String tableName = "catalog.custom_schema.my_table";
+        String actual = MappingHelper.externalName(tableName);
+        Assertions.assertThat(actual).isEqualTo("\"catalog\".\"custom_schema\".\"my_table\"");
+    }
+
+    @Test
+    public void tableName_to_externalName_withQuotes() {
+        String tableName = "custom_schema.\"my_table\"";
+        String actual = MappingHelper.externalName(tableName);
+        Assertions.assertThat(actual).isEqualTo("\"custom_schema\".\"my_table\"");
+    }
+
+    @Test
+    public void tableName_to_externalName_withEscapedQuotes() {
+        String tableName = "custom_\"\"schema.\"my_\"\"table\"";
+        String actual = MappingHelper.externalName(tableName);
+        Assertions.assertThat(actual).isEqualTo("\"custom_\"\"schema\".\"my_\"\"table\"");
+    }
+
+    @Test
+    public void tableName_to_externalName_withMoreEscapedQuotes() {
+        String tableName = "custom_\"\"\"\"schema.\"my_\"\"table\"\"\"";
+        String actual = MappingHelper.externalName(tableName);
+        Assertions.assertThat(actual).isEqualTo("\"custom_\"\"\"\"schema\".\"my_\"\"table\"\"\"");
+    }
+
+    @Test
+    public void tableName_to_externalName_with2Quotes() {
+        String tableName = "\"custom_schema\".\"my_table\"";
+        String actual = MappingHelper.externalName(tableName);
+        Assertions.assertThat(actual).isEqualTo("\"custom_schema\".\"my_table\"");
+    }
+
+    @Test
+    public void tableName_to_externalName_with3Quotes() {
+        String tableName = "\"catalog\".\"custom_schema\".\"my_table\"";
+        String actual = MappingHelper.externalName(tableName);
+        Assertions.assertThat(actual).isEqualTo("\"catalog\".\"custom_schema\".\"my_table\"");
```

Comment:
nitty nit: static import for assertThat :)

Reply:
fixed: https://github.com/hazelcast/hazelcast/pull/24332/commits/9cba4f43615d9a3d1c3861427b256387f92f1d9f